ipsource.go

```diff
@@ -323,6 +330,11 @@ func (c Command) GetIPs(ctx context.Context, versions IPVersions) ([]net.IP, err
 
 	err := cmd.Run()
 
+	exitCode := cmd.ProcessState.ExitCode()
+	if exitCode != 0 {
+		return nil, fmt.Errorf("Process exit with: %d", exitCode)
+	}
```

Comment:
Probably want to report the stderr content with `c.logger.Error()`

I suggest adding some `c.logger.Debug()` throughout as well in relevant spots.